ledger-core/virtual/integration/deduplication/constructor_test.go

```diff
@@ -139,3 +141,206 @@ func TestConstructor_SamePulse(t *testing.T) {
+			if test.VState.ProvidedContent != nil {
+				test.VState.ProvidedContent.OrderedQueue = []reference.Global{outgoing}
+			}
+
+			if test.VFindCall != nil {
+				test.VFindCall.Callee = outgoing
+				test.VFindCall.Outgoing = outgoing
+			}
+
+			if test.VDelegatedCall != nil {
+				test.VDelegatedCall.Callee = outgoing
+				test.VDelegatedCall.CallOutgoing = outgoing
+			}
+
+			{
+				requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+				requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+				executionMock := runnerMock.AddExecutionMock(outgoing.String())
+				executionMock.AddStart(nil, &execution.Update{
+					Type:   execution.Done,
+					Result: requestResult,
+				})
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+				assert.Equal(t, []byte("123"), result.ReturnArguments)
+				return false
+			})
+			typedChecker.VStateRequest.Set(func(request *payload.VStateRequest) bool {
+				VStateReportDone := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+				server.SendPayload(ctx, &test.VState)
+				testutils.WaitSignalsTimed(t, 10*time.Second, VStateReportDone)
+
+				assert.Equal(t, object, request.Object)
+
+				return false
+			})
+			typedChecker.VFindCallRequest.Set(func(request *payload.VFindCallRequest) bool {
+				if test.VFindCall == nil {
+					t.Fatal("unreachable")
+				}
+
+				assert.Equal(t, outgoing, request.Outgoing)
+				assert.Equal(t, object, request.Callee)
+
+				test.VFindCall.LookedAt = request.LookAt
+				server.SendPayload(ctx, test.VFindCall)
+
+				return false
+			})
+			typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
+				return false
+			})
+
+			server.IncrementPulseAndWaitIdle(ctx)
+
+			if test.VStateSendBefore {
+				VStateReportDone := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+				server.SendPayload(ctx, &test.VState)
+				testutils.WaitSignalsTimed(t, 10*time.Second, VStateReportDone)
+			}
+
+			if test.VDelegatedCall != nil {
+				server.SendPayload(ctx, test.VDelegatedCall)
+			}
+
+			{
+				pl := payload.VCallRequest{
+					CallType:       payload.CTConstructor,
+					CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+					Callee:         class,
+					CallSiteMethod: "New",
+					CallOutgoing:   outgoing,
+				}
+
+				server.SendPayload(ctx, &pl)
+			}
+
+			{
+				testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
+				testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+			}
+
+			{
+				if test.VStateSendBefore {
+					assert.Equal(t, 0, typedChecker.VStateRequest.Count())
+				} else {
+					assert.Equal(t, 1, typedChecker.VStateRequest.Count())
+				}
+
+				if test.VFindCall != nil && test.VFindCall.CallResult != nil {
+					assert.Equal(t, 1, typedChecker.VCallResult.Count())
+				}
```

Comment:
this is wrong math. Put callResultExpected (or similar) as boolean flag into the tests table

Reply:
I came here for the same comment - add expected results into the test table

Reply:
fixed